### PR TITLE
[docs] Update IP autodetection page

### PR DESCRIPTION
**Pull request by lmm**
## Description

This PR updates the [IP autodetection page](https://projectcalico.docs.tigera.io/networking/ip-autodetection) to include instructions for operator-based installations.

Deploy preview: https://deploy-preview-5499--calico-master.netlify.app/networking/ip-autodetection

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: calico/networking/ip-autodetection.md [hunk 1]
```diff
@@ -50,39 +50,92 @@ By default, {{site.prodname}} uses the **first-found** method; the first valid I
 - Address used by the node to reach a particular IP or domain (**can-reach**)
 - Regex to include matching interfaces (**interface**)
 - Regex to exclude matching interfaces (**skip-interface**)
+- A list of IP ranges in CIDR format to determine valid IP addresses on the node to choose from (**cidrs**)
 - Address assigned to kubernetes node (**kubernetes-internal-ip**)
 
+> **Note**: `kubernetes-internal-ip` is not available via the [Installation API]({{site.baseurl}}/reference/installation/api#operator.tigera.io/v1.Installation).
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:56]: This is available now - we might just need to pull in the latest CRDs and API reference docs from the operator repo.

File: calico/networking/ip-autodetection.md [hunk 2]
```diff
@@ -50,39 +50,92 @@ By default, {{site.prodname}} uses the **first-found** method; the first valid I
...
+
+  ```
+  kind: Installation
+  apiVersion: operator.tigera.io/v1
+  metadata:
+    name: default
+  spec:
+    calicoNetwork:
+      nodeAddressAutodetectionV4:
+		skipInterface: eth.*
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:118]: indentation seems off here

File: calico/networking/ip-autodetection.md [hunk 3]
```diff
@@ -140,7 +203,40 @@ In the following scenarios, you may want to configure a specific IP and subnet:
 - Changes to cross subnet packet encapsulation
 - Changes to host IP address
 
-You can configure specific IP address and subnet for a node using environment variables or by updating the [Node resource]({{ site.baseurl }}/reference/resources/node).
+{% tabs %}
+  <label:Operator,active:true>
+<%
+##### Configure IP and subnet using default Installation resource
+
+To configure the IP addresses used by the node, edit the default [Installation]({{site.baseurl}}/reference/installation/api#operator.tigera.io/v1.Installation)
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:211]: This section isn't yet supported by the operator (and may not be, if there's no need for it).

I suggest adding a note to that extent within this tab:

> Manually specifying IP addresses for individual nodes is not supported when installing via the operator. Please use one of the supported IP autodetection methods above instead.

File: calico/networking/ip-autodetection.md (hunk 1 above)
Comment (lmm) [reply calico/networking/ip-autodetection.md]: Fixed. I'll update the API reference docs in a follow-up PR once I've gotten the `build-operator-reference` target sorted out.

File: calico/networking/ip-autodetection.md [hunk 4]
```diff
@@ -50,39 +50,105 @@ By default, {{site.prodname}} uses the **first-found** method; the first valid I
...
+  <label:Operator,active:true>
+<%
+As noted previously, the default autodetection method is **first valid interface found** (first-found). To use a different autodetection method, edit the default [Installation]({{site.baseurl}}/reference/installation/api#operator.tigera.io/v1.Installation) custom resource, specifying the method. Below are examples of the supported autodetection methods:
 
-##### Using environment variables and node resource
+> **Note**: To configure the default autodetection method for IPv6 for any of the below methods, use the field `nodeAddressAutodetectionV6`.
+{: .alert .alert-info}
 
-Because you can configure IP address and subnet using either environment variables or node resource, the following table describes how values are synchronized.
+- **IP or domain name**
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:73]: I'd call this "Source address used to reach an IP or domain name" - this isn't specifying an address to use, rather it's telling the node "Use the IP selected by the kernel when routing to the given address"

File: calico/networking/ip-autodetection.md [hunk 5]
```diff
@@ -50,39 +50,105 @@ By default, {{site.prodname}} uses the **first-found** method; the first valid I
...
+  spec:
+    calicoNetwork:
+      nodeAddressAutodetectionV4:
+        cidrs:
+          - "192.168.200.0/24"
+  ```
 
+- **Kubernetes Node IP**
+
+  An internal IP address assigned to the Kubernetes node.
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:136]: I think we should explicitly call out that this comes from the `Node.Status.Addresses` field

Comment (caseydavenport) [reply calico/networking/ip-autodetection.md:136]: And maybe move this one to the top - I think this is probably the most often desired auto-detection method.

File: calico/networking/ip-autodetection.md [hunk 6]
```diff
@@ -109,28 +175,38 @@ Where autodetection methods are based on:
...
-  An IP address assigned to kubernetes node (INTERNAL-IP)
+  An internal IP address assigned to the Kubernetes node.
 
   ```
   kubectl set env daemonset/calico-node -n kube-system IP_AUTODETECTION_METHOD=kubernetes-internal-ip
   ```
 
+- **Including CIDRs**
+
+  A list of IP ranges in CIDR format to determine valid IP addresses on the node to choose from. For example:
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:202]: "Calico will select any IP address from the node that falls within the given CIDRs"

File: calico/networking/ip-autodetection.md [hunk 7]
```diff
@@ -50,39 +50,105 @@ By default, {{site.prodname}} uses the **first-found** method; the first valid I
...
+    name: default
+  spec:
+    calicoNetwork:
+      nodeAddressAutodetectionV4:
+        interface: eth.*
+  ```
+
+- **Excluding matching interfaces**
+
+  A regular expression in Golang syntax that excludes interfaces that match. For example:
```
Comment (caseydavenport) [inline calico/networking/ip-autodetection.md:105]: I think for all of these a slightly more explicit wording would be helpful.

e.g., "Calico will choose an address on each node from an interface that does not match the given regex"

File: calico/networking/ip-autodetection.md (hunk 5 above)
Comment (lmm) [reply calico/networking/ip-autodetection.md:136]: I put that detail in the node configuration reference page but wasn't sure if it should go in this page or not. I think you're right though

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**